lib/subscription.js

```diff
@@ -6,7 +6,7 @@ const { kHooks } = require('./symbols')
 const SubscriptionConnection = require('./subscription-connection')
 const { getProtocolByName } = require('./subscription-protocol')
 
-function createConnectionHandler ({ subscriber, fastify, onConnect, onDisconnect, lruGatewayResolvers, entityResolversFactory, subscriptionContextFn, keepAlive, fullWsTransport }) {
+function createConnectionHandler ({ subscriber, fastify, onConnect, onDisconnect, entityResolversFactory, subscriptionContextFn, keepAlive, fullWsTransport }) {
```

Comment:
How is this functionality implemented in the gateway?

Reply:
Instead of being added to the `context` is passed as parameter to the `makeResolver` function where it's used.

https://github.com/mercurius-js/mercurius-gateway/blob/main/lib/gateway/build-gateway.js#L120
https://github.com/mercurius-js/mercurius-gateway/blob/main/lib/gateway/make-resolver.js#L453

Reply:
While checking this stuff I found an error in the gateway, the values is not passed in the refresh action. Pr is coming

Reply:
as @jonnydgreen mentioned in another comment the hooks as is are not working at all. The validation is done by mercusiur, then once removed the `gateway` hooks is not possible to add it.


I will add a hook management internal to the gateway, that should be called with:

```
  app.graphql.gateway.addHook('onSomeGatewayHook', async (execution, context) => {
    ...
  })
```

WDYT? @mcollina @jonnydgreen

Reply:
And once we have defined the right approach I'll update docs and examples properly